### PR TITLE
Laboratory work #1, Varvara Migacheva - 22FPL2

### PR DESCRIPTION
**Pull request by varmig2000**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1 @@
+print("Hello, world!")
```
Comment (artyomtugaryov) [inline main.py:1]: Please, remove the file

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Nothing to review

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -11,6 +11,12 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if type(text) != str:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:14]: use isinstance

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -11,6 +11,12 @@ def tokenize(text: str) -> list[str] | None:
...
+        return None
+    f = filter(str.isalpha, text)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:16]: can you make it in one line?

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -19,6 +25,19 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if type(tokens) != list:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:28]: .

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -19,6 +25,19 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+        return None
+    for i in tokens:
+        if type(i) != str:
+            return None
+    dict = {}
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:33]: bad name - do not use built in names

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -28,6 +47,13 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if type(language) != str or type(text) != str:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:50]: .

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Do not foget to change score

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (varmig2000) [reply lab_1_classify_by_unigrams/main.py:16]: i don't think i can...... can you explain? sorry, i forgot to ask you about it...

Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:16]: I absolutelly sure that you can write filter + join + lower + casting to list

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -11,6 +11,10 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    text1 = list(''.join(filter(str.isalpha, text)).lower())
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:16]: Perfecto!

Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:16]: But do not create the variable

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -28,6 +45,13 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    dictionary = {
+    "name": language,
+    "freq": calculate_frequencies(tokenize(text))
+        }
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:53]: :(

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -14,8 +13,7 @@ def tokenize(text: str) -> list[str] | None:
     """
     if not isinstance(text, str):
         return None
-
-    return [token.lower() for token in text if token.isalpha()]
+    return list(''.join(filter(str.isalpha, text)).lower())
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:16]: silicon-valley-jared

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -24,13 +22,19 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
     """
-    if not (
-            isinstance(tokens, list) and all(isinstance(el, str) for el in tokens)
-    ):
+    if not isinstance(tokens, list):
         return None
-
-    frequency_dict = {el: (tokens.count(el) / len(tokens)) for el in tokens}
-    return frequency_dict
+    for i in tokens:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:27]: please use only one for in this function

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**